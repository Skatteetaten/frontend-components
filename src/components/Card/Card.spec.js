import React from 'react';
import toJson from 'enzyme-to-json';
import { shallow, mount } from 'enzyme';
import { Card, CardColor } from '../index';

function oppsettShallow(props) {
  return shallow(<Card {...props} />);
}

function oppsettFullDOM(props) {
  return mount(<Card {...props} />);
}

describe('Card komponent', () => {
  it('matcher snapshot', () => {
    const wrapper = oppsettFullDOM();
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('rendrer Card med default props', () => {
    const wrapper = oppsettFullDOM();

    expect(wrapper.prop('titlesize')).toEqual('x-large');
    expect(wrapper.prop('color')).toEqual('neutralGrey');
    expect(wrapper.prop('marginbottom')).toEqual('2px');
    expect(wrapper.prop('expand')).toEqual(false);
    expect(wrapper.prop('isExpanded')).toEqual(true);
    expect(wrapper.prop('title')).toEqual(undefined);
  });

  it('setter Card med riktige props', () => {
    const wrapper = oppsettFullDOM({
      color: CardColor.BEIGE,
      title: 'Ikke fullført arbeidsoppgave',
      titlesize: 'large',
      marginbottom: '10px',
      expand: true,
      isExpanded: false,
      id: 'card-id',
      className: 'card-classname',
    });

    expect(wrapper.prop('titlesize')).toEqual('large');
    expect(wrapper.prop('color')).toEqual('beige');
    expect(wrapper.prop('marginbottom')).toEqual('10px');
    expect(wrapper.prop('title')).toEqual('Ikke fullført arbeidsoppgave');
    expect(wrapper.prop('isExpanded')).toEqual(false);
    expect(wrapper.prop('expand')).toEqual(true);
    expect(wrapper.prop('id')).toEqual('card-id');
    expect(wrapper.prop('className')).toEqual('card-classname');
  });

  it('endrer state når utvid knapp klikkes', () => {
    const wrapper = oppsettShallow({
      expand: true,
      isExpanded: true,
    });
    const expandBtn = wrapper.find('button');

    expect(wrapper.html()).toContain('titleExpand');
    expect(wrapper.html()).toContain('expandIcon');
    expect(wrapper.state('isExpandedState')).toEqual(true);
    expandBtn.simulate('click');
    expect(wrapper.state('isExpandedState')).toEqual(false);
  });
});
