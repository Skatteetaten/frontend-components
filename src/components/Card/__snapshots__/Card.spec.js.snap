// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card komponent matcher snapshot 1`] = `
<Card
  buttonType="button"
  color="beige"
  expand={false}
  isExpanded={true}
  margin="medium"
  marginbottom="2px"
  titleTagName="div"
  titlesize="x-large"
>
  <div
    className=

        {
          background-color: #f9ede2;
          border: none;
          box-sizing: border-box;
          color: rgba(0, 0, 0, 0.95);
          display: flow-root;
          margin-bottom: 2px;
          padding-bottom: 0px;
          padding-left: 0px;
          padding-right: 0px;
          padding-top: 0px;
        }
        &:hover {
          cursor: default;
        }
        &:focus {
          outline: none;
        }
  >
    <div
      className=

          {
            animation-name: keyframes from{opacity:0;}to{opacity:1;};
            font-size: 1rem;
            margin-bottom: 12px;
            margin-left: 12px;
            margin-right: 12px;
            margin-top: 12px;
            text-align: left;
          }
    />
  </div>
</Card>
`;
